astropy/io/fits/util.py

```diff
@@ -982,3 +984,74 @@ def _get_array_mmap(array):
         if isinstance(base.base, mmap.mmap):
             return base.base
         base = base.base
+
+
+def get_posargs(function):
+    """Helper function to retrieve the positional parameters from a
+    function
+
+    inspect.getargspec is deprecated in Python 3, while
+    inspect.signature does not exist in Python 2. This function caters
+    for both.
+
+    """
+    try:
+        posargs = list(inspect.signature(function).parameters.keys())
```

Comment:
Please use `astropy.utils.compat.funcsigs.signature` instead of this try/except.